### PR TITLE
ROC AUC metric implementation

### PR DESCRIPTION
**Pull request by mzuevx**
Wrapper to use  sklearn.metrics.roc_auc_score in accuracy_checker.
Needed for DLRM MLPERF 100GB model validation.

### CODE REVIEW

**Comment by eaidova** [comment]
Jenkins ok to test

File: tools/accuracy_checker/accuracy_checker/metrics/classification.py [hunk 1]
```diff
@@ -262,3 +266,26 @@ def reset(self):
...
         self.fp = 0
         self.fn = 0
+
+class RocAucScore(PerImageEvaluationMetric):
+    __provider__ = 'roc_auc_score'
+    annotation_types = (ClassificationAnnotation, TextClassificationAnnotation)
+    prediction_types = (ClassificationPrediction, ArgMaxClassificationPrediction)
+
+    def configure(self):
+        self.reset()
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/classification.py:276]: please add check that  roc_auc_score is not None

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/metrics/classification.py:276]: and please add new metric to the readme

Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/metrics/classification.py:278]: Fixed

**Review by eaidova — APPROVED** [review]
